### DISCUSSION TITLE
How to send actions to the server from workflow-extension file

### DISCUSSION THREAD

**Discussion by rileydanejohnston** [eclipse-glsp/glsp#927] [General]
Hello,
How can I send actions/operations directly to the server from workflow-extension file? I want to be able to listen for changes made directly to my source model, then trigger the gmodel-factory so the changes are reflected in the graph. I see glspVscodeConnector.sendActionToActiveClient(), but I’m wondering if it’s possible to send the actions/operations directly to the server because my GLSP client will not always be the active/focused panel (we have two separate vscode extensions sharing the same client window).

**Comment by tortmayr** [comment]
Hi @rileydanejohnston,
what server flavor are you using (Java or Node)? 
For Theia+Java GLSP already offers a [`SourceModelWatcher` API](https://github.com/eclipse-glsp/glsp-server/blob/master/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/sourcemodelwatcher/SourceModelWatcher.java) that provide exactly the feature that your describe (listen for external source model changes & notify the client). 
However, I'm not sure whether this is currently also supported in the vscode integration and/or the node server. 
We have to look into that.
